Files openbabel-2.1.1-orig/.ChangeLog.swp and openbabel-2.1.1/.ChangeLog.swp differ
diff -N -r -u openbabel-2.1.1-orig/CMakeLists.txt openbabel-2.1.1/CMakeLists.txt
--- openbabel-2.1.1-orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/CMakeLists.txt	2007-11-19 20:51:49.461250000 +0100
@@ -0,0 +1,88 @@
+project(openabel)
+set(BABEL_MAJ_VER   2)
+set(BABEL_MIN_VER   1)
+set(BABEL_PATCH_VER 1)
+if(WIN32)
+  set(BABEL_LIBRARY   openbabel)
+else(WIN32)
+  set(BABEL_LIBRARY   openbabel)
+endif(WIN32)
+set(BABEL_VERSION  "${BABEL_MAJ_VER}.${BABEL_MIN_VER}.${BABEL_PATCH_VER}")
+set(BABEL_DATADIR  "{CMAKE_INSTALL_PREFIX}/share/openbabel")
+set(LIB            "lib")   ## FIXME: add lib suffix!
+
+option(WITH_INCHI   "build inchi lib" ON)
+option(BUILD_SHARED "enable static build support" ON)
+if(BUILD_SHARED)
+  set(BUILD_TYPE SHARED)
+else(BUILD_SHARED)
+  set(BUILD_TYPE STATIC)
+endif(BUILD_SHARED)
+
+if(WIN32)
+  set(MODULE_EXTENSION ".dll")
+else(WIN32)
+  set(MODULE_EXTENSION ".so")
+endif(WIN32)
+
+# configure checks
+find_package(LibXml2)
+find_package(ZLIB)
+if(ZLIB_FOUND)
+  add_definitions(-DHAVE_LIBZ)
+  include_directories(${ZLIB_INCLUDE_DIR})
+endif(ZLIB_FOUND)
+
+include(CheckIncludeFile)
+include(CheckIncludeFileCXX)
+include(CheckSymbolExists)
+include(CheckTypeSize)
+include(CheckCSourceCompiles)
+
+check_include_file(conio.h      HAVE_CONIO_H)
+check_include_file(sys/time.h   HAVE_SYS_TIME_H)
+check_include_file(time.h       HAVE_TIME_H)
+check_include_file_cxx(sstream  HAVE_SSTREAM)
+
+check_symbol_exists(rint          "math.h"     HAVE_RINT)
+check_symbol_exists(snprintf      "stdio.h"    HAVE_SNPRINTF)
+check_symbol_exists(sranddev      "stdlib.h"   HAVE_SRANDDEV)
+check_symbol_exists(strcasecmp    "string.h"   HAVE_STRCASECMP)
+check_symbol_exists(strncasecmp   "string.h"   HAVE_STRNCASECMP)
+
+set(CMAKE_EXTRA_INCLUDE_FILES time.h)
+check_type_size(clock_t CLOCK_T)
+
+if(NOT WIN32)
+  SET(CMAKE_REQUIRED_FLAGS -Werror)
+  check_c_source_compiles("
+   #include <sys/types.h>
+   #include <dirent.h>
+    int main(){
+     extern int matchFiles (struct dirent *entry_p);
+     struct dirent **entries_pp;
+     int count = scandir (\"./\", &entries_pp, matchFiles, 0);
+     return 0;
+    }
+   " SCANDIR_NEEDS_CONST)
+
+   set(OB_MODULE_PATH ${CMAKE_INSTALL_PREFIX}/${LIB}/openbabel)
+endif(NOT WIN32)
+
+configure_file(${CMAKE_SOURCE_DIR}/src/config.h.cmake ${CMAKE_BINARY_DIR}/include/openbabel/babelconfig.h)
+install(FILES ${CMAKE_BINARY_DIR}/include/openbabel/babelconfig.h DESTINATION include/openbabel-2.0/openbabel)
+configure_file(${CMAKE_SOURCE_DIR}/openbabel-2.0.pc.cmake ${CMAKE_BINARY_DIR}/openbabel-2.0.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/openbabel-2.0.pc DESTINATION ${LIB}/pkgconfig)
+
+include_directories(${CMAKE_SOURCE_DIR}/include
+                         ${CMAKE_SOURCE_DIR}/data
+                         ${CMAKE_BINARY_DIR}/include
+)
+
+add_subdirectory(include)
+add_subdirectory(data)
+add_subdirectory(doc)
+add_subdirectory(src)
+add_subdirectory(test)
+add_subdirectory(tools)
+
diff -N -r -u openbabel-2.1.1-orig/data/CMakeLists.txt openbabel-2.1.1/data/CMakeLists.txt
--- openbabel-2.1.1-orig/data/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/data/CMakeLists.txt	2007-11-19 19:25:53.523750000 +0100
@@ -0,0 +1,11 @@
+
+set(to_install
+    aromatic.txt bondtyp.txt isotope.txt resdata.txt
+    atomtyp.txt element.txt isotope-small.txt phmodel.txt
+    patterns.txt torlib.txt types.txt
+    SMARTS_InteLigand.txt
+    ghemical.prm
+    logp.txt psa.txt mr.txt
+)
+
+install(FILES ${to_install} DESTINATION share/openbabel/${BABEL_VERSION})
diff -N -r -u openbabel-2.1.1-orig/doc/CMakeLists.txt openbabel-2.1.1/doc/CMakeLists.txt
--- openbabel-2.1.1-orig/doc/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/doc/CMakeLists.txt	2007-11-19 19:25:53.523750000 +0100
@@ -0,0 +1,2 @@
+file(GLOB man_1 *.1)
+install(FILES ${man_1} DESTINATION share/man/man1)
diff -N -r -u openbabel-2.1.1-orig/include/CMakeLists.txt openbabel-2.1.1/include/CMakeLists.txt
--- openbabel-2.1.1-orig/include/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/include/CMakeLists.txt	2007-11-19 19:25:53.539375000 +0100
@@ -0,0 +1,8 @@
+
+file(GLOB openbabel_headers openbabel/*.h)
+file(GLOB math_headers      openbabel/math/*.h)
+set(openbabel_headers ${openbabel_headers} chemdrawcdx.h)
+
+install(FILES inchi_api.h          DESTINATION include/inchi)
+install(FILES ${openbabel_headers} DESTINATION include/openbabel-2.0/openbabel)
+install(FILES ${math_headers}      DESTINATION include/openbabel-2.0/openbabel/math)
diff -N -r -u openbabel-2.1.1-orig/include/openbabel/obmolecformat.h openbabel-2.1.1/include/openbabel/obmolecformat.h
--- openbabel-2.1.1-orig/include/openbabel/obmolecformat.h	2007-02-26 17:46:15.000000000 +0100
+++ openbabel-2.1.1/include/openbabel/obmolecformat.h	2007-11-19 19:25:53.617500000 +0100
@@ -110,7 +110,7 @@
   static OBMol* MakeCombinedMolecule(OBMol* pFirst, OBMol* pSecond);
   //@}
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
   typedef stdext::hash_map<std::string, unsigned> NameIndexType;
 #else
   typedef std::map<std::string, unsigned> NameIndexType;
diff -N -r -u openbabel-2.1.1-orig/openbabel-2.0.pc.cmake openbabel-2.1.1/openbabel-2.0.pc.cmake
--- openbabel-2.1.1-orig/openbabel-2.0.pc.cmake	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/openbabel-2.0.pc.cmake	2007-11-19 19:25:53.633125000 +0100
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/@LIB@
+includedir=${prefix}/include
+pkgincludedir=${includedir}/openbabel-2.0
+
+Name: Open Babel library
+Description: libopenbabel
+Version: @BABEL_VERSION@
+Libs: -L${libdir} -lopenbabel
+Cflags: -I${pkgincludedir}
diff -N -r -u openbabel-2.1.1-orig/src/CMakeLists.txt openbabel-2.1.1/src/CMakeLists.txt
--- openbabel-2.1.1-orig/src/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/src/CMakeLists.txt	2007-11-19 22:58:19.336250000 +0100
@@ -0,0 +1,87 @@
+
+
+set(fingerprints_srcs
+    fingerprints/finger2.cpp
+    fingerprints/finger3.cpp
+)
+
+set(forcefields_srcs
+    forcefields/forcefieldghemical.cpp
+)
+
+set(math_srcs
+    math/matrix3x3.cpp
+    math/vector3.cpp
+)
+
+set(openbabel_srcs
+    atom.cpp
+    base.cpp
+    bitvec.cpp
+    bond.cpp
+    bondtyper.cpp
+    canon.cpp
+    chains.cpp
+    chiral.cpp
+    data.cpp
+    fingerprint.cpp
+    forcefield.cpp
+    generic.cpp
+    grid.cpp
+    kekulize.cpp
+    matrix.cpp
+    mol.cpp
+    molchrg.cpp
+    obconversion.cpp
+    oberror.cpp
+    obiter.cpp
+    obutil.cpp
+    parsmart.cpp
+    patty.cpp
+    phmodel.cpp
+    rand.cpp
+    residue.cpp
+    ring.cpp
+    rotor.cpp
+    rotamer.cpp
+    tokenst.cpp
+    typer.cpp
+    transform.cpp
+    groupcontrib.cpp
+)
+
+if(BUILD_SHARED)
+  if(WIN32)
+    set(openbabel_srcs ${openbabel_srcs}
+        dlhandler_win32.cpp
+        )
+    add_definitions(${BABEL_LIBRARY} -DUSING_DYNAMIC_LIBS)
+  else(WIN32)
+    set(openbabel_srcs ${openbabel_srcs}
+        dlhandler_unix.cpp
+        )
+  endif(WIN32)
+endif(BUILD_SHARED)
+
+add_library(${BABEL_LIBRARY} ${BUILD_TYPE}
+                ${openbabel_srcs}
+                ${fingerprints_srcs}
+                ${forcefields_srcs}
+                ${math_srcs}
+)
+
+if(NOT WIN32)
+  set(libs ${libs} m)
+endif(NOT WIN32)
+
+target_link_libraries(${BABEL_LIBRARY} ${libs} ${LIBXML2_LIBRARIES} ${ZLIB_LIBRARY})
+set_target_properties(${BABEL_LIBRARY} PROPERTIES
+                                            VERSION ${BABEL_VERSION}
+                                            SOVERSION ${BABEL_MAJ_VER})
+install(TARGETS ${BABEL_LIBRARY}
+                RUNTIME DESTINATION bin
+                LIBRARY DESTINATION ${LIB}
+                ARCHIVE DESTINATION ${LIB}
+)
+
+add_subdirectory(formats)
diff -N -r -u openbabel-2.1.1-orig/src/config.h.cmake openbabel-2.1.1/src/config.h.cmake
--- openbabel-2.1.1-orig/src/config.h.cmake	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/src/config.h.cmake	2007-11-19 19:25:53.711250000 +0100
@@ -0,0 +1,160 @@
+/* src/config.h.in.  Generated from configure.in by autoheader.  */
+
+/* Where the data files are located */
+#define BABEL_DATADIR "@BABEL_DATADIR@"
+
+/* The version of Open Babel */
+#define BABEL_VERSION "@BABEL_VERSION@"
+
+/* Used to export symbols for DLL / shared library builds */
+#if defined(WIN32)
+ #if defined(USING_OBDLL) // e.g. in src/main.cpp
+  #define EXTERN __declspec(dllimport) extern
+ #else
+  #define EXTERN __declspec(dllexport) extern
+ #endif
+#else //Everything else (behaviour as original)
+ #define EXTERN extern
+#endif
+
+/* The file extension used for shared modules */
+#define MODULE_EXTENSION "@MODULE_EXTENSION@"
+
+/* Used to export symbols for DLL / shared library builds */
+#if defined(WIN32)
+ #if defined(USING_OBDLL) // e.g. in src/main.cpp
+  #define OBAPI __declspec(dllimport)
+ #else
+  #define OBAPI __declspec(dllexport)
+ #endif
+#else //Everything else (behaviour as original)
+ #define OBAPI
+#endif
+
+
+/* Used to export symbols for DLL / shared library builds */
+#if defined(WIN32)
+ #if defined(USING_OBDLL) // e.g. in src/main.cpp
+  #define OBCOMMON __declspec(dllimport)
+ #else
+  #define OBCOMMON __declspec(dllexport)
+ #endif
+#else //Everything else (behaviour as original)
+ #define OBCOMMON
+#endif
+
+
+/* Used to export symbols for DLL / shared library builds */
+#if defined(WIN32)
+ #if defined(USING_OBDLL) // e.g. in src/main.cpp
+  #define OBCONV __declspec(dllimport)
+ #else
+  #define OBCONV __declspec(dllexport)
+ #endif
+#else //Everything else (behaviour as original)
+ #define OBCONV
+#endif
+
+
+/* Used to export symbols for DLL / shared library builds */
+#if defined(WIN32)
+ #if defined(USING_OBDLL) // e.g. in src/main.cpp
+  #define OBERROR __declspec(dllimport)
+ #else
+  #define OBERROR __declspec(dllexport)
+ #endif
+#else //Everything else (behaviour as original)
+ #define OBERROR
+#endif
+
+
+/* Used to export symbols for DLL / shared library builds */
+#if defined(WIN32)
+ #if defined(USING_OBDLL) // e.g. in src/main.cpp
+  #define OBFPTR __declspec(dllimport)
+ #else
+  #define OBFPTR __declspec(dllexport)
+ #endif
+#else //Everything else (behaviour as original)
+ #define OBFPTR
+#endif
+
+#ifdef _MSC_VER
+ // Supress warning on deprecated functions
+ #pragma warning(disable : 4996)
+ // Supress warning that compiler is ignoring C++ exception specification
+ #pragma warning( disable : 4290 )
+ // Supress warning on signed/unsigned comparison with < or > (harmless, but maybe should be fixed)
+ #pragma warning( disable : 4018 )
+ // Supress warning on unreferenced formal parameter
+ #pragma warning( disable : 4100 )
+ //
+ #pragma warning( disable : 4251 )
+
+
+ #include <crtdbg.h>
+
+ #ifdef _DEBUG
+ #define DEBUG_NEW new(_NORMAL_BLOCK, THIS_FILE, __LINE__)
+ #else
+  #define DEBUG_NEW new
+ #endif
+#endif  // _MSC_VER
+/* have <conio.h> */
+#cmakedefine HAVE_CONIO_H 1
+
+/* have <sys/time.h> */
+#cmakedefine HAVE_SYS_TIME_H 1
+
+/* have <time.h> */
+#cmakedefine HAVE_TIME_H 1
+
+/* have <sstream> */
+#cmakedefine HAVE_SSTREAM 1
+
+/* have symbol clock_t */
+#cmakedefine HAVE_CLOCK_T 1
+
+/* have symbol rint */
+#cmakedefine HAVE_RINT 1
+
+/* have symbol snprintf */
+#cmakedefine HAVE_SNPRINTF 1
+
+/* have symbol sranddev */
+#cmakedefine HAVE_SRANDDEV 1
+
+/* have symbol strcasecmp */
+#cmakedefine HAVE_STRCASECMP 1
+
+/* have symbol strncasecmp */
+#cmakedefine HAVE_STRNCASECMP 1
+
+/* have struct clock_t */
+#cmakedefine HAVE_CLOCK_T 1
+
+#if defined(WIN32)
+ #ifndef HAVE_SNPRINTF
+  #define snprintf _snprintf
+  #define HAVE_SNPRINTF 1
+ #endif
+
+ #ifndef HAVE_STRCASECMP
+  #define strcasecmp _stricmp
+  #define HAVE_STRCASECMP 1
+ #endif
+
+ #ifndef HAVE_STRNCASECMP
+  #define strncasecmp _strnicmp
+  #define HAVE_STRNCASECMP 1
+ #endif
+#endif  // WIN32
+
+#cmakedefine SCANDIR_NEEDS_CONST 1
+#ifdef SCANDIR_NEEDS_CONST
+ #define SCANDIR_CONST const
+#else
+ #define SCANDIR_CONST
+#endif
+
+#define OB_MODULE_PATH "@OB_MODULE_PATH@"
diff -N -r -u openbabel-2.1.1-orig/src/fingerprints/finger3.cpp openbabel-2.1.1/src/fingerprints/finger3.cpp
--- openbabel-2.1.1-orig/src/fingerprints/finger3.cpp	2007-07-05 06:26:17.000000000 +0200
+++ openbabel-2.1.1/src/fingerprints/finger3.cpp	2007-11-19 19:25:53.711250000 +0100
@@ -18,6 +18,11 @@
 #include <openbabel/babelconfig.h>
 #include <openbabel/mol.h>
 
+#ifdef HAVE_SSTREAM
+#include <sstream>
+#else
+#include <strstream>
+#endif
 #include <fstream>
 #include <map>
 #include <string>
diff -N -r -u openbabel-2.1.1-orig/src/formats/CMakeLists.txt openbabel-2.1.1/src/formats/CMakeLists.txt
--- openbabel-2.1.1-orig/src/formats/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/src/formats/CMakeLists.txt	2007-11-19 21:23:29.898750000 +0100
@@ -0,0 +1,154 @@
+if(WITH_INCHI)
+  add_definitions(-DINCHI_LINK_AS_DLL)
+  add_subdirectory(inchi)
+endif(WITH_INCHI)
+
+set(SIMPLEFORMATS
+    APIInterface
+    copyformat
+    rxnformat
+)
+
+set(OBMOLFORMATS
+    CSRformat
+    PQSformat
+    alchemyformat
+    acrformat
+    amberformat
+    balstformat
+    bgfformat
+    boxformat
+    cacaoformat
+    cacheformat
+    carformat
+    cccformat
+    chem3dformat
+    chemdrawct
+    chemdrawcdx
+    chemtoolformat
+    cifformat
+    crkformat
+    cssrformat
+    dmolformat
+#    fastsearchformat
+    fastaformat
+    fchkformat
+    featformat
+    fhformat
+#    fingerprintformat
+    freefracformat
+    gamessformat
+    gaussformat
+    ghemicalformat
+    gromos96format
+    hinformat
+    jaguarformat
+    mdlformat
+    mmodformat
+    mol2format
+    molreport
+    mopacformat
+    mpdformat
+    mpqcformat
+    nwchemformat
+    pcmodelformat
+    pdbformat
+    povrayformat
+    qchemformat
+    reportformat
+    shelxformat
+    smilesformat
+    thermoformat
+    tinkerformat
+    titleformat
+    turbomoleformat
+    unichemformat
+    viewmolformat
+    xedformat
+    xyzformat
+    yasaraformat
+    zindoformat
+)
+
+if(WITH_INCHI)
+  set(formats
+    ${formats}
+    getinchi
+    inchiformat
+  )
+  set(libs ${libs} inchi)
+endif(WITH_INCHI)
+
+if(LIBXML2_FOUND)
+  include_directories(${LIBXML2_INCLUDE_DIR})
+  set(xml_formats
+    cdxmlformat
+    cmlformat
+    cmlreactlformat
+    pubchem
+    xmlformat
+  )
+endif(LIBXML2_FOUND)
+
+if(NOT WIN32)
+  set(libs ${libs} m)
+endif(NOT WIN32)
+
+if(BUILD_SHARED)
+  if(WIN32)
+    set(openbabel_srcs ${openbabel_srcs}
+        dlhandler_win32
+        )
+  else(WIN32)
+    set(openbabel_srcs ${openbabel_srcs}
+        dlhandler_unix
+        )
+  endif(WIN32)
+endif(BUILD_SHARED)
+
+add_library( fastsearchformat ${BUILD_TYPE} fastsearchformat.cpp ../fingerprint.cpp )
+target_link_libraries(fastsearchformat ${libs} ${BABEL_LIBRARY})
+install(TARGETS fastsearchformat
+              RUNTIME DESTINATION bin
+              LIBRARY DESTINATION ${LIB}/openbabel
+              ARCHIVE DESTINATION ${LIB}/openbabel)
+set_target_properties(fastsearchformat PROPERTIES OUTPUT_NAME fastsearchformat PREFIX "")
+
+add_library( fingerprintformat ${BUILD_TYPE}  fingerprintformat.cpp ../fingerprint.cpp obmolecformat.cpp )
+target_link_libraries( fingerprintformat ${libs} ${BABEL_LIBRARY})
+install(TARGETS fingerprintformat
+              RUNTIME DESTINATION bin
+              LIBRARY DESTINATION ${LIB}/openbabel
+              ARCHIVE DESTINATION ${LIB}/openbabel)
+set_target_properties( fingerprintformat PROPERTIES OUTPUT_NAME fingerprintformat PREFIX "")
+
+foreach(format ${SIMPLEFORMATS})
+  add_library(${format} ${BUILD_TYPE} ${format}.cpp)
+  target_link_libraries(${format} ${libs} ${BABEL_LIBRARY})
+  install(TARGETS ${format}
+                  RUNTIME DESTINATION bin
+                  LIBRARY DESTINATION ${LIB}/openbabel
+                  ARCHIVE DESTINATION ${LIB}/openbabel)
+  set_target_properties(${format} PROPERTIES OUTPUT_NAME ${format} PREFIX "")
+endforeach(format)
+
+foreach(format ${OBMOLFORMATS})
+  add_library(${format} ${BUILD_TYPE} ${format}.cpp obmolecformat.cpp)
+  target_link_libraries(${format} ${libs} ${BABEL_LIBRARY})
+  install(TARGETS ${format}
+                  RUNTIME DESTINATION bin
+                  LIBRARY DESTINATION ${LIB}/openbabel
+                  ARCHIVE DESTINATION ${LIB}/openbabel)
+  set_target_properties(${format} PROPERTIES OUTPUT_NAME ${format} PREFIX "")
+endforeach(format)
+
+foreach(format ${xml_formats})
+  add_library(${format} ${BUILD_TYPE} xml/${format}.cpp xml/xml.cpp obmolecformat.cpp)
+  target_link_libraries(${format} ${libs} ${BABEL_LIBRARY})
+  install(TARGETS ${format}
+                  RUNTIME DESTINATION bin
+                  LIBRARY DESTINATION ${LIB}/openbabel
+                  ARCHIVE DESTINATION ${LIB}/openbabel
+  )
+  set_target_properties(${format} PROPERTIES OUTPUT_NAME ${format} PREFIX "")
+endforeach(format)
diff -N -r -u openbabel-2.1.1-orig/src/formats/inchi/CMakeLists.txt openbabel-2.1.1/src/formats/inchi/CMakeLists.txt
--- openbabel-2.1.1-orig/src/formats/inchi/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/src/formats/inchi/CMakeLists.txt	2007-11-19 19:25:53.805000000 +0100
@@ -0,0 +1,25 @@
+project(inchi)
+set(INCHI_MAJ_VER   0)
+set(INCHI_MIN_VER   0)
+set(INCHI_PATCH_VER 1)
+
+set(inchi_srcs
+  ichi_bns.c ichican2.c ichicano.c ichicans.c ichiisot.c
+  ichilnct.c ichimak2.c ichimake.c ichimap1.c ichimap2.c
+  ichimap4.c ichinorm.c ichiparm.c ichiprt1.c ichiprt2.c
+  ichiprt3.c ichiqueu.c ichiread.c ichiring.c ichirvr1.c
+  ichirvr2.c ichirvr3.c ichirvr4.c ichirvr5.c ichirvr6.c
+  ichirvr7.c ichisort.c ichister.c ichitaut.c inchi_dll.c
+  inchi_dll_main.c runichi.c strutil.c util.c
+)
+include_directories(${CMAKE_SOURCE_DIR}/include/inchi)
+add_library(inchi SHARED ${inchi_srcs})
+set_target_properties(inchi PROPERTIES DEFINE_SYMBOL _USRDLL)
+set_target_properties(inchi PROPERTIES
+                                 VERSION ${INCHI_MAJ_VER}.${INCHI_MIN_VER}.${INCHI_PATCH_VER}
+                                 SOVERSION ${INCHI_MAJ_VER})
+install(TARGETS inchi
+                inchi DESTINATION bin
+                LIBRARY DESTINATION lib
+                ARCHIVE DESTINATION lib
+)
diff -N -r -u openbabel-2.1.1-orig/src/formats/inchiformat.cpp openbabel-2.1.1/src/formats/inchiformat.cpp
--- openbabel-2.1.1-orig/src/formats/inchiformat.cpp	2007-07-07 08:35:53.000000000 +0200
+++ openbabel-2.1.1/src/formats/inchiformat.cpp	2007-11-19 19:25:53.898750000 +0100
@@ -21,7 +21,11 @@
 #include <openbabel/obmolecformat.h>
 
 #include "inchi_api.h"
+#ifdef HAVE_SSTREAM
 #include <sstream>
+#else
+#include <strstream>
+#endif
 #include <set>
 #include <vector>
 #include "openbabel/chiral.h"
diff -N -r -u openbabel-2.1.1-orig/src/formats/xml/cmlformat.cpp openbabel-2.1.1/src/formats/xml/cmlformat.cpp
--- openbabel-2.1.1-orig/src/formats/xml/cmlformat.cpp	2007-04-14 02:34:13.000000000 +0200
+++ openbabel-2.1.1/src/formats/xml/cmlformat.cpp	2007-11-19 19:25:53.930000000 +0100
@@ -23,6 +23,11 @@
 #ifdef WIN32
 #pragma warning (disable : 4800)
 #endif
+#ifdef HAVE_SSTREAM
+#include <sstream>
+#else
+#include <strstream>
+#endif
 
 using namespace std;
 namespace OpenBabel
diff -N -r -u openbabel-2.1.1-orig/src/obconversion.cpp openbabel-2.1.1/src/obconversion.cpp
--- openbabel-2.1.1-orig/src/obconversion.cpp	2007-07-01 21:16:17.000000000 +0200
+++ openbabel-2.1.1/src/obconversion.cpp	2007-11-19 19:25:54.133125000 +0100
@@ -423,6 +423,8 @@
   //////////////////////////////////////////////////////
   bool OBConversion::SetOutFormat(OBFormat* pOut)
   {
+    if(pOut==NULL)
+      return true;
     pOutFormat=pOut;
     return !(pOutFormat->Flags() & NOTWRITABLE);
   }
diff -N -r -u openbabel-2.1.1-orig/test/CMakeLists.txt openbabel-2.1.1/test/CMakeLists.txt
--- openbabel-2.1.1-orig/test/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/test/CMakeLists.txt	2007-11-19 19:25:54.133125000 +0100
@@ -0,0 +1,8 @@
+
+add_executable(roundtrip roundtrip.cpp)
+target_link_libraries(roundtrip ${BABEL_LIBRARY})
+install(TARGETS roundtrip
+                RUNTIME DESTINATION bin
+                LIBRARY DESTINATION lib
+                ARCHIVE DESTINATION lib
+)
diff -N -r -u openbabel-2.1.1-orig/tools/CMakeLists.txt openbabel-2.1.1/tools/CMakeLists.txt
--- openbabel-2.1.1-orig/tools/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ openbabel-2.1.1/tools/CMakeLists.txt	2007-11-19 19:25:54.133125000 +0100
@@ -0,0 +1,21 @@
+set(tools
+      babel
+      obchiral
+      obenergy
+      obfit
+      obgrep
+      obminimize
+      obprop
+      obrotamer
+      obrotate
+)
+
+foreach(tool ${tools})
+  add_executable(${tool} ${tool}.cpp)
+  target_link_libraries(${tool} ${BABEL_LIBRARY})
+  install(TARGETS ${tool}
+                  RUNTIME DESTINATION bin
+                  LIBRARY DESTINATION lib
+                  ARCHIVE DESTINATION lib
+  )
+endforeach(tool)
diff -N -r -u openbabel-2.1.1-orig/tools/obgrep.cpp openbabel-2.1.1/tools/obgrep.cpp
--- openbabel-2.1.1-orig/tools/obgrep.cpp	2007-01-05 17:29:56.000000000 +0100
+++ openbabel-2.1.1/tools/obgrep.cpp	2007-11-19 21:53:01.398750000 +0100
@@ -28,7 +28,7 @@
 #include <openbabel/obconversion.h>
 #include <openbabel/parsmart.h>
 
-#ifdef _WIN32
+#ifdef MSC_VER
 	typedef char TCHAR;
 	#include "XGetOpt.h"
 #else
